### PR TITLE
externalize sd_notify and require only when needed

### PR DESCRIPTION
**Pull request by seuros**
With this change, systemd integration will load only if sd_notify gem is present in the Gemfile.

- sd_notify is is the same gem used by puma.
- We do not need to test it here anymore.
- sd_notify can be loaded only in the watchdog thread.
- sd_notify is optional.

### CODE REVIEW

**Comment by mperham** [comment]
I don’t like forcing the user to add the gem in order to use systemd. Is there an ENV we can use when running in systemd to conditionally load s/sd_notify?

**Comment by seuros** [comment]
The gem has to be added in puma to use the  type=notify in the service.

On Jruby, sidekiq should have the type=normal.

ping @agis

File: lib/sidekiq/cli.rb [hunk 1]
```diff
@@ -417,5 +417,5 @@ def rails_app?
   end
 end
 
-require "sidekiq/systemd"
+require "sidekiq/systemd" if Gem::Specification.find_all_by_name("sd_notify").any?
```
Comment (mperham) [inline lib/sidekiq/cli.rb:420]: Too specific. What if the user wants to supply their own version or use a forked gem? Just catch LoadError like usual?

Comment (seuros) [reply lib/sidekiq/cli.rb:420]: They can require the file manualy. 

The only requirement is that is should be named sd_notify.rb and SdNotify

**Comment by mperham** [comment]
I think you missed the parts in `bin/sidekiq`. Honestly though I'm not seeing much value with this change. I don't want the user to have to add another gem to gain systemd support. I forked `lib/sidekiq/sd_notify.rb` years ago; it's very simple and has required zero maintenance.

**Pull request closed without merging**